models/redhat.go

```diff
@@ -187,56 +186,60 @@ type RedhatPackageState struct {
 }
 
 // ConvertRedhat :
-func ConvertRedhat(cveJSONs []RedhatCVEJSON) (cves []RedhatCVE, err error) {
-	for _, cve := range cveJSONs {
-		details := []RedhatDetail{}
-		for _, d := range cve.Details {
-			d = util.TrimSpaceNewline(d)
-			details = append(details, RedhatDetail{Detail: d})
-		}
-
-		references := []RedhatReference{}
-		for _, r := range cve.References {
-			r = util.TrimSpaceNewline(r)
-			references = append(references, RedhatReference{Reference: r})
-		}
-
-		cve.Bugzilla.Description = util.TrimSpaceNewline(cve.Bugzilla.Description)
-		cve.Statement = util.TrimSpaceNewline(cve.Statement)
-
-		var publicDate time.Time
-		if cve.PublicDate != "" {
-			if strings.HasSuffix(cve.PublicDate, "Z") {
-				publicDate, err = time.Parse("2006-01-02T15:04:05Z", cve.PublicDate)
-			} else {
-				publicDate, err = time.Parse("2006-01-02T15:04:05", cve.PublicDate)
-			}
-			if err != nil {
-				return nil, xerrors.Errorf("Failed to parse date. date: %s err: %w", cve.PublicDate, err)
+func ConvertRedhat(cveJSONs iter.Seq2[RedhatCVEJSON, error]) iter.Seq2[RedhatCVE, error] {
+	return func(yield func(RedhatCVE, error) bool) {
+		for cve, err := range cveJSONs {
+			if err != nil && !yield(RedhatCVE{}, err) {
+				return
 			}
```

Comment:
This is a discussion rather than suggestion. 

```suggestion
			if err != nil {
				if yield(RedhatCVE{}, err) {
					continue
				} else {
					return
				}
			}
```

How about this? My concern is using "cve" below these lines in spite of err != nil.

Reply:
As you pointed out, if the cve is not appropriate, such as when json decoding fails, there is no need to put it in subsequent iterations.

fixed: https://github.com/vulsio/gost/compare/07d45f5ed762ede8400f26b1d8f710159be049fe..df01ebc4deb4fe69854a0d55bc1b7403ca7001a9